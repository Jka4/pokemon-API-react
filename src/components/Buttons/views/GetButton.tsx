/* eslint-disable import/no-anonymous-default-export */
import React, { useCallback } from 'react';

import { useSelector, useDispatch } from 'react-redux';

import { Button } from '@material-ui/core';
import Badge from '@material-ui/core/Badge';
import CasinoIcon from '@material-ui/icons/Casino';

import { PokesTypes } from 'types/index';

interface GetButtonProps {
  randomPokemon: PokesTypes[];
  pokemonArr: PokesTypes[];
}

const GetButton: React.FC = () => {
  const randomPokemon = useSelector((state: GetButtonProps) => state.randomPokemon);
  const pokemonArr = useSelector((state: GetButtonProps) => state.pokemonArr);

  const dispatch = useDispatch();

  const getRandomPokemon = useCallback(() => {
    const randomPokemon = pokemonArr[Math.floor(1 + Math.random() * pokemonArr.length)];
    dispatch({ type: 'ADD_RANDOM_POKEMON', payload: randomPokemon });
  }, [dispatch, pokemonArr]);

  return (
    <>
      <Badge color="secondary" badgeContent={randomPokemon.length}>
        <Button
          onClick={getRandomPokemon}
          className="getPokemon"
          variant="contained"
          color="secondary"
          endIcon={<CasinoIcon />}
        >
          GET A RANDOM POKEMON
        </Button>
      </Badge>
    </>
  );
};

export default GetButton;
