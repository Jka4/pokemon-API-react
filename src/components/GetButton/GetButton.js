import React from "react";
import store from "@Store";
import { getRandomPokemon } from "@APIutils";
import { NavLink } from "react-router-dom";

import './styles/style.scss';


const GetButton = () => {
  const clearState = () => {
    store.dispatch({ type: "CLEAR_RANDOM_POKEMON" });
  };

  return (
    <div className='Button'>
      <button id='getPokemon' onClick={() => getRandomPokemon(806)}>
        GET A RANDOM POKEMON
        </button>

      <button id='clear' onClick={clearState}>
        CLEAR
        </button>

      <NavLink to={`/allPokemons/`} >
        <button id='getAllPokemon' >
          GET ALL POKEMON
        </button>
      </NavLink>
    </div>
  );
}

export default GetButton;
