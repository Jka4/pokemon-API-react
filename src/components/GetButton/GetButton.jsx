/* eslint-disable import/no-anonymous-default-export */
import React from "react";
import { Provider, connect } from "react-redux";
import { NavLink } from "react-router-dom";

import { Button } from "@material-ui/core";
import Badge from "@material-ui/core/Badge";
import CasinoIcon from "@material-ui/icons/Casino";

import { getRandomPokemon } from "@APIutils";
import store from "@Store";

import "./styles/style.scss";

const GetButton = ({
  randomPokemons = [],
  randomPokemonsFetching = Boolean,
}) => {
  const clearState = () => {
    store.dispatch({ type: "CLEAR_RANDOM_POKEMON" });
  };

  return (
    <div className="Button">
      <NavLink to={`/allPokemons/`}>
        <Button variant="contained" color="secondary">
          {" "}
          SHOW ALL POKEMON{" "}
        </Button>
      </NavLink>

      <Button
        onClick={clearState}
        className="clearBtn"
        variant="contained"
        color="secondary"
      >
        CLEAR
      </Button>

      <Badge color="secondary" badgeContent={randomPokemons.length}>
        <Button
          onClick={() => getRandomPokemon()}
          id="getPokemons"
          className="getPokemons"
          variant="contained"
          color="secondary"
          endIcon={!randomPokemonsFetching && <CasinoIcon />}
        >
          {randomPokemonsFetching ? "Loading..." : "GET A RANDOM POKEMON"}
        </Button>
      </Badge>
    </div>
  );
};

const ConnectedGetButton = connect((store) => {
  return {
    randomPokemons: store.randomPokemons,
    randomPokemonsFetching: store.randomPokemonsFetching,
  };
})(GetButton);

export default (props = {}) => (
  <Provider store={store}>
    <ConnectedGetButton {...props} />
  </Provider>
);
