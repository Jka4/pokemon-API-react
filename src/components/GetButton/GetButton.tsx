/* eslint-disable import/no-anonymous-default-export */
import React from 'react';
import { NavLink } from 'react-router-dom';

import { Provider, connect } from 'react-redux';
import store from 'Store/store';

import { Button } from '@material-ui/core';
import Badge from '@material-ui/core/Badge';
import CasinoIcon from '@material-ui/icons/Casino';

import './styles/style.scss';

import { Pokes, IStoreType } from 'types';

interface GetButtonProps {
  randomPokemon: Pokes[];
  pokemonArr: Pokes[];
}

const GetButton: React.FC<GetButtonProps> = ({ randomPokemon = [], pokemonArr = [] }: GetButtonProps) => {
  const clearState = () => {
    store.dispatch({ type: 'CLEAR_RANDOM_POKEMON' });
  };

  const handleClick = () => {
    const randomPokemonFromArr = pokemonArr[Math.floor(1 + Math.random() * pokemonArr.length)];
    store.dispatch({ type: 'ADD_RANDOM_POKEMON', randomPokemonFromArr });
  };

  return (
    <div className="Button">
      <NavLink to={`/allPokemon/`}>
        <Button variant="contained" color="secondary">
          {' '}
          SHOW ALL POKEMON{' '}
        </Button>
      </NavLink>

      <Button onClick={clearState} className="clearBtn" variant="contained" color="secondary">
        CLEAR
      </Button>

      <Badge color="secondary" badgeContent={randomPokemon.length}>
        <Button
          onClick={handleClick}
          id="getPokemon"
          className="getPokemon"
          variant="contained"
          color="secondary"
          endIcon={<CasinoIcon />}
        >
          GET A RANDOM POKEMON
        </Button>
      </Badge>
    </div>
  );
};

const ConnectedGetButton = connect((store: IStoreType) => {
  return {
    randomPokemon: store.randomPokemon,
    pokemonArr: store.pokemonArr,
  };
})(GetButton);

export default (props = {}) => (
  <Provider store={store}>
    <ConnectedGetButton {...props} />
  </Provider>
);
