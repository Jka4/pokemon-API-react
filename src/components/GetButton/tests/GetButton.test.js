import React from 'react';
import GetButton from '@GetButton';
import renderer from 'react-test-renderer';

it('renders correctly', async () => {
	const tree = renderer
		.create(<GetButton />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
