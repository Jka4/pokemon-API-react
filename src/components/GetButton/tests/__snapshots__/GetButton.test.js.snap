// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Button"
>
  <button
    id="getPokemon"
    onClick={[Function]}
  >
    GET A RANDOM POKEMON
  </button>
  <button
    id="clear"
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
